Mathlib/Data/Int/Cast/Basic.lean

```diff
@@ -95,10 +95,13 @@ theorem cast_negOfNat (n : ℕ) : ((negOfNat n : ℤ) : R) = -n := by simp [Int.
 @[simp, norm_cast]
 theorem cast_add : ∀ m n, ((m + n : ℤ) : R) = m + n
   | (m : ℕ), (n : ℕ) => by simp [-Int.natCast_add, ← Int.natCast_add]
-  | (m : ℕ), -[n+1] => by erw [cast_subNatNat, cast_natCast, cast_negSucc, sub_eq_add_neg]
-  | -[m+1], (n : ℕ) => by
-    erw [cast_subNatNat, cast_natCast, cast_negSucc, sub_eq_iff_eq_add, add_assoc,
-      eq_neg_add_iff_add_eq, ← Nat.cast_add, ← Nat.cast_add, Nat.add_comm]
+  | (m : ℕ), -[n+1] =>
+    show (subNatNat m n.succ : R) = _ by
```

Comment:
I'm still not too psyched about this. Yes, the other branches do the same, but I'm not convinced they are also a form of technical debt...